draft-parecki-oauth-client-id-metadata-document.md

```diff
@@ -182,6 +182,18 @@ by requiring the `token_endpoint_auth_method` property be set to `"none"`.
 
 TBD: We may want a property such as `client_id_expires_at` for indicating that the client is ephemeral and not valid after a given timestamp, especially for documents issued by a service for development purposes.
 
+## Client Metadata Documents for Development Purposes {#documents_for_development}
+
+When developing applications against an authorization server which uses this specification, developers often encounter the issue of "how do I serve a Client ID Metadata Document at a publicly accessible https URL whilst developing my application on my localhost?".
+
+To enable developers to author applications on their machines, without exposing their machines to the public internet, the usage of Client ID Metadata Document Services by the authorization server is RECOMMENDED.
+
+A Client ID Metadata Document Service is a web service through which developers can acquire a stable URL to a Client ID Metadata Document. This service MAY expire clients from time to time, and MAY require developers to provide additional information about the client being developed.
+
+An authorization server MAY place additional restrictions on the `redirect_uris`, for instance, limiting them to the same-origin as the `client_id` or `client_uri` properties, however, if an authorization server does place additional restrictions on the `redirect_uris` then it SHOULD provide at least one Client ID Metadata Document Service which is exempt from these restrictions.
```

Comment:
I'm not sure I follow this sentence. I think I'm getting confused about "at least one Client ID Metadata Document Service", what is "service" here? I thought the concept of the AS hosting metadata for in-development clients was the service.

Reply:
It is a bit awkward to phrase correctly. Essentially what I'm trying to say is "if you want to restrict redirect_uri's, then have a service where developers can register/request a Client ID Metadata Document for use in non-production use cases", essentially a trusted static registry which serves multiple Client ID Metadata Documents from the one domain.

Reply:
oh I get it now. I was reading that as restricting redirect URLs in the context of dev clients. I think that would make more sense in the very beginning of this section to set the context for why this metadata hosting service is needed.